### ISSUE TITLE
cargo-lipo doesn't work with workspaces

### ISSUE THREAD

**Issue by poros** [TimNN/cargo-lipo#5]
I have a project which defines a cargo workspace as per http://doc.crates.io/manifest.html#the-workspace-section

I'd like to invoke `cargo lipo` on one of the members of my workspace (it's a sub-crate exposing C bindings for the main crate), but both `cargo lipo -p MEMBER` or `cargo lipo --all` are missing.

Is there any plan to make `cargo-lipo` support workspaces?

**Comment by TimNN** [comment]
Hi @poros,

there haven’t been any plans until now ;)

I can look into it over the next 1 - 2 weeks, however if you want to start an implementation, I’ll gladly accept a pull request (just comment here when you start work, so we don’t duplicate work).

Please ping me if you haven’t heard back in ~2 weeks, that means this issue has probably dropped off my radar (I’m not currently actively using `cargo-lipo`).

**Comment by poros** [comment]
Ping ;)

**Comment by TimNN** [comment]
@poros: Sorry for dropping the ball on this. I've just published `cargo-lipo 2.0.0-beta.1` which has some basic support for workspaces: You can `cd` into the directory of the sub-crate and successfully execute `cargo-lipo` from there.

There are also some more concrete plans to do a full 2.0 release with support for all workspace features & flags.

**Comment by poros** [comment]
No worries, it's what I have been doing so far in my Makefile. I'm looking forward to the 0.2 release! :)

**Comment by chrisballinger** [comment]
Just ran into this problem myself. :)

**Comment by regexident** [comment]
I'm getting this when trying to install the latest beta (which strangely enough [is listed on crates.io](https://crates.io/crates/cargo-lipo/2.0.0-beta-3)) via `cargo install`:

```bash
cargo install cargo-lipo "2.0.0-beta-3"
    Updating registry `https://github.com/rust-lang/crates.io-index`
  Installing cargo-lipo v1.0.3
   <snip>
   Compiling cargo-lipo v1.0.3
    Finished release [optimized] target(s) in 22.96s
  Installing <snip>/.cargo/bin/cargo-lipo
error: could not find `2.0.0-beta-3` in registry `https://github.com/rust-lang/crates.io-index`
     Summary Successfully installed cargo-lipo! Failed to install 2.0.0-beta-3 (see error(s) above).
```

Any idea what's going wrong here? 🤔

**Comment by TimNN** [comment]
I think the syntax you want is `cargo install --version 2.0.0-beta-3 cargo-lipo`, now that you already installed the older version probably with `--force` as well. (Your command is attempting to install a crate called "2.0.0-beta-3", I believe.

**Comment by regexident** [comment]
That fixed it, thanks!

**Comment by TimNN** [comment]
I've released a `2.0.0` which includes the workspace changes, so I think this issue can be closed.
